src/PeriodActions.php

```diff
@@ -8,51 +8,50 @@
 
 trait PeriodActions
 {
-    public function forLastMonth(): static
+    private const UNIT_HOURS = 'hours';
+    private const UNIT_DAYS = 'days';
+    private const UNIT_WEEKS = 'weeks';
+    private const UNIT_MONTH = 'month';
+    private const UNIT_YEAR = 'year';
+    private const UNIT_DECADE = 'decade';
+    private const DEFAULT_TIME_UNIT_DURATION = 1;
+
+    /**
+     * Generic method to set a period based on the starting point and the duration type.
+     */
+    private function setPeriodFromDuration($duration, string $unit): static
```

Comment:
Could you type `$duration`?

Reply:
Also, as this package would be extended by other people, I would suggest to use `protected` accessor here.